### PR TITLE
Speed up CI

### PR DESCRIPTION
**Pull request by marlass**
Currently running pipeline takes more than 10 minutes.
We can use stages/jobs feature to split each task we do in `build.sh` script and run them in parallel.
I prepared this quick PoC and the time already dropped to 6 minutes.
We can go even further and also cache dependencies installs and so on.

This PR is not ready and the point of it is to just start discussion about that issue.
Let me know what you think :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+    - develop
+    - "/^feature\\/.*$/"
+cache: yarn
+jobs:
+  include:
+    - script: './ci-scripts/e2e-tests.sh'
+    - script: './ci-scripts/other.sh'
+    - script: './ci-scripts/unit-tests.sh'
+    - script: './ci-scripts/lint.sh'
+# before_script:
```
Comment (hackergil) [inline .travis.yml]: Please get rid of commented code

File: ci-scripts/unit-tests.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+
+echo "Updating dependencies"
+yarn
+echo "-----"
```
Comment (hackergil) [inline ci-scripts/unit-tests.sh]: Please remove this echo

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
 branches:
   only:
-  - develop
-  - "/^feature\\/.*$/"
-before_script:
-- npm install -g angular-cli
-script: "./build.sh"
+    - develop
+    - "/^feature\\/.*$/"
+cache: yarn
```
Comment (hackergil) [inline .travis.yml]: I believe @plabadie had some concerns about adding this. Can you check with him?

**Comment by marlass** [comment]
I fixed issues pointed out in code review. I removed the cache directive after talk with @plabadie
Could you check it again? @hackergil

**Comment by hackergil** [comment]
Hey @marlass . I took another look. I noticed that build.sh is still there. Can you please remove the logic from it and just add calls to the other scripts? This is so that we can still execute it locally (even if not in a parallel way but sequential) and we don't have duplicated logic for it.

**Comment by marlass** [comment]
Changed `build.sh` script.

**Pull request merged**